src/OrchardCore.Modules/OrchardCore.Workflows/Http/Scripting/HttpMethodsProvider.cs

```diff
@@ -134,18 +136,99 @@ public HttpMethodsProvider(IHttpContextAccessor httpContextAccessor)
                                  select new JProperty(param.Key, JArray.FromObject(param.Value.ToArray()))).ToArray()))
             };
 
+            // This should be deprecated
             _requestFormAsJsonMethod = new GlobalMethod
             {
                 Name = "requestFormAsJson",
                 Method = serviceProvider => (Func<JObject>)(() =>
                     new JObject((from field in httpContextAccessor.HttpContext.Request.Form
                                  select new JProperty(field.Key, JArray.FromObject(field.Value.ToArray()))).ToArray()))
             };
+
+            _requestDataAsJsonObjectMethod = new GlobalMethod
+            {
+                Name = "requestDataAsJsonObject",
+                Method = serviceProvider => (Func<JObject>)(() =>
+                {
+                    JObject result = null;
+
+                    if(httpContextAccessor.HttpContext != null)
+                    {
+                        if (httpContextAccessor.HttpContext.Request.HasFormContentType)
+                        {
+                            try
+                            {
+                                result = new JObject((from field in httpContextAccessor.HttpContext.Request.Form
+                                        select new JProperty(field.Key, JArray.FromObject(field.Value.ToArray()))).ToArray());
```

Comment:
I would suggest to differentiate between arrays and simple properties, the resulting object is much cleaner.

Reply:
Yes, I need to fix this part here still. Felt asleep. I think, since requests can send multiple values for a single form element that it makes sense that these are represented as an array. Though, when the request is sent as JSON using a single property is allowed. So this method needs more validation on the actual JSON passed so that the resulting object be always using the same "schema".

Reply:
Ok, I see what you mean now that I look at this : 
https://github.com/StatCan/StatCan.OrchardCore/blob/ae8543049fbe3e1e93864ae6dcb6ac60e8668584/src/Modules/StatCan.OrchardCore.Scripting/FormsGlobalMethodsProvider.cs#L53-L58

Sanitizing might also be a good thing.

Reply:
Yeah, I think sanitizing should be a option on the globalMethod and be true by default, but can be disabled per use case.

Reply:
I don't understand why sanitization is necessary here. Or make it available in the script if the user wants to invoke it.

Reply:
You could pass `<script>something<script>` as a form value to inject a script. That's the reason why I think it should be sanitized. Or maybe we should not do it here and somewhere else. Making it optional and default to true makes sense if we want this. Else I can remove, let me know.